### ISSUE TITLE
Code 128 improperly decoded as UPC_E when image is at angle

### ISSUE THREAD

**Issue by crawdell** [micjahn/ZXing.Net#50]
I have found that a Code 128 barcode image is improperly decoded as UPC_E when the image is at an angle but a second image of the same barcode is properly decoded as Code 128.  Note the images below.  The angled image returns the incorrect barcode format and value but the approx. 90 degree image works fine.

Correct Code 128 value:
X000WDQ1V3

Improperly decoded value as UPC_E:
00522047

Note that this happens in my own code but also is the same issue in the WindowsFormsDemo application that comes with this distribution.  I've tried forcing Code 128 as the only format but then it just fails to decode.  

The defaults for the WindowsFormsDemo are checked:
enable Auto Rotate
enable Try Harder
enable Try Inverted
![a__3a91](https://user-images.githubusercontent.com/23389296/31741623-c3720a00-b409-11e7-8836-d7720fdfcb24.jpg)

![a__2b5b](https://user-images.githubusercontent.com/23389296/31741635-cdb62a50-b409-11e7-818b-f191fa5a3dd0.jpg)

**Comment by micjahn** [comment]
Zxing doesn't guarantee to find every barcode. Especially if the barcode is rotated the library has some issues.
You should restrict the formats which are looked for.
```
var reader = new BarcodeReader();
reader.Options.PossibleFormats = new List<BarcodeFormat> { BarcodeFormat.CODE_128};
```
It prevents that you will find false positives.
Try to capture the barcodes unrotated, if possible, use an continuesly image stream.
It improves the user experience.
The zxing library itself was originally developed for mobile device scenarios which continuesly scanning for codes while the user focuses the camera.

Feel free to provide a PR which improves the scanning result for rotated codes.

**Comment by crawdell** [comment]
Thanks for the feedback.  I did restrict the format to Code 128 but then the decoding would fail on the angled version.  

In my testing this Zxing library has done a good job with angled images, just not with Code 128.  I've tried other online barcode decoders and they don't have any problem with the angled image I've been testing with so I think there could be room for improvement.  I'll post a PR specific to Code 128 since I've not seen issues in my testing with other formats. 

Unfortunately in my use case I will only have still images to work from that have been previously captured so I need the best decoding I can get from imperfect images.
